core/src/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -2085,82 +2085,87 @@ public Long getInteger( Object object ) throws KettleValueException {
+            case STORAGE_TYPE_INDEXED:
+              return convertStringToBigNumber( (String) index[( (Integer) object ).intValue()] );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+        case TYPE_INTEGER:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return BigDecimal.valueOf( ( (Long) object ).longValue() );
+            case STORAGE_TYPE_BINARY_STRING:
+              return BigDecimal.valueOf( ( (Long) convertBinaryStringToNativeType( (byte[]) object ) ).longValue() );
+            case STORAGE_TYPE_INDEXED:
+              return BigDecimal.valueOf( ( (Long) index[( (Integer) object ).intValue()] ).longValue() );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+        case TYPE_NUMBER:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return BigDecimal.valueOf( ( (Double) object ).doubleValue() );
+            case STORAGE_TYPE_BINARY_STRING:
+              return BigDecimal.valueOf( ( (Double) convertBinaryStringToNativeType( (byte[]) object ) ).doubleValue() );
+            case STORAGE_TYPE_INDEXED:
+              return BigDecimal.valueOf( ( (Double) index[( (Integer) object ).intValue()] ).doubleValue() );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+        case TYPE_DATE:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return convertDateToBigNumber( (Date) object );
+            case STORAGE_TYPE_BINARY_STRING:
+              return convertDateToBigNumber( (Date) convertBinaryStringToNativeType( (byte[]) object ) );
+            case STORAGE_TYPE_INDEXED:
+              return convertDateToBigNumber( (Date) index[( (Integer) object ).intValue()] );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+        case TYPE_BOOLEAN:
+          switch ( storageType ) {
+            case STORAGE_TYPE_NORMAL:
+              return convertBooleanToBigNumber( (Boolean) object );
+            case STORAGE_TYPE_BINARY_STRING:
+              return convertBooleanToBigNumber( (Boolean) convertBinaryStringToNativeType( (byte[]) object ) );
+            case STORAGE_TYPE_INDEXED:
+              return convertBooleanToBigNumber( (Boolean) index[( (Integer) object ).intValue()] );
+            default:
+              throw new KettleValueException( toString() + " : Unknown storage type " + storageType + " specified." );
+          }
+        case TYPE_BINARY:
+          throw new KettleValueException( toString() + " : I don't know how to convert binary values to integers." );
```

Comment:
Please clean up these copy/paste errors. It shouldn't say "... to integers." as this is a  BigDecimal. It should say "I don't know how to convert ... to BigDecimals."